prov/psm/src/psmx_init.c

```diff
@@ -450,11 +450,25 @@ struct fi_provider psmx_prov = {
 	.cleanup = psmx_fini
 };
 
-static int psmx_get_int_env(char *name, int default_value)
+static void psmx_reg_and_read(const char *name, const char *help_string,
+			char **value)
+{
+	*value = NULL;
+
+	if (fi_var_register(&psmx_prov, name, help_string))
+		FI_INFO(&psmx_prov, FI_LOG_CORE,
+			"Failed to register name variable\n", name);
+	if (fi_var_get(&psmx_prov, name, value))
+		FI_INFO(&psmx_prov, FI_LOG_CORE,
+			"Failed to read uuid variable\n");
```

Comment:
We should be able to avoid these FI_INFO calls in each provider by moving them into the fi_var_* calls.